crates/libs/wallet/src/types/compact_wad.rs

```diff
@@ -124,27 +126,28 @@ impl<U: Unit + DeserializeOwned> FromStr for CompactWad<U> {
 
 impl<U: Unit + Serialize> fmt::Display for CompactWads<U> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        use serde::ser::Error;
-        let mut data = Vec::new();
-        ciborium::into_writer(self, &mut data).map_err(|e| fmt::Error::custom(e.to_string()))?;
-        let encoded = general_purpose::URL_SAFE.encode(data);
-        write!(f, "{}{}", CASHU_PREFIX, encoded)
+        let mut tokens = Vec::new();
+        for wad in &self.0 {
+            tokens.push(wad.to_string());
+        }
+        write!(f, "{}", tokens.join(":"))
     }
 }
 
 impl<U: Unit + DeserializeOwned> FromStr for CompactWads<U> {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let s = s
-            .strip_prefix(CASHU_PREFIX)
-            .ok_or(Error::UnsupportedWadFormat)?;
+        let token_strings: Vec<&str> = s.split(':').collect();
 
-        let decode_config = general_purpose::GeneralPurposeConfig::new()
-            .with_decode_padding_mode(bitcoin::base64::engine::DecodePaddingMode::Indifferent);
-        let decoded = GeneralPurpose::new(&alphabet::URL_SAFE, decode_config).decode(s)?;
-        let token = ciborium::from_reader(&decoded[..])?;
-        Ok(token)
+        let mut wads = Vec::with_capacity(token_strings.len());
+        for token_str in token_strings {
```

Comment:
you are collecting twice here when you actually just need to do it once:

```rust

let wads = s.split(':')
   .map(|tokent_str| CompactWad::from_str(token_str))
   .collect::<Result<Vec<_>, _>>()
   .map_err(|e| Error::InvalidWadToken(Box::new(e)))?;;

        Ok(CompactWads(wads))
```

should work